components/datetime/src/options/length.rs

```diff
@@ -126,6 +127,7 @@ impl Default for Bag {
 /// [`Element dateFormats`]: https://unicode.org/reports/tr35/tr35-dates.html#dateFormats
 /// [`DateTimeFormat`]: super::super::DateTimeFormat
 #[derive(Debug, Clone, Copy, PartialEq)]
+#[allow(clippy::exhaustive_enums)] // this type is stable
```

Comment:
Nit: This should be `non_exhaustive` because we may want to add additional datetime styles.